File: [283]移动零.cpp
#include <vector>
using namespace std;

class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int p = 0;
        for (int num : nums) {
            if (num != 0)
                nums[p++] = num;
        }
        for (; p < nums.size(); ++p)
            nums[p] = 0;
    }
};

File: [283]移动零.py
from typing import List


class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        nonzero_count = 0
        for i, num in enumerate(nums):
            if num != 0:
                nums[nonzero_count] = num
                nonzero_count += 1
        
        for i in range(nonzero_count, len(nums)):
            nums[i] = 0
        
        return nums
